src/helpers/github.ts

```diff
@@ -99,3 +99,66 @@ export async function isCiGreen({ octokit, logger, env }: Context, sha: string,
     return false;
   }
 }
+
+function parseTarget({ payload }: Context, target: string) {
+  const owner = payload.repository.owner || "";
+  const [orgParsed, repoParsed] = target.split("/");
+  let repoTarget = null;
+  if (orgParsed !== owner) {
+    return null;
+  } else if (repoParsed) {
+    repoTarget = repoParsed;
+  }
+  return { org: owner, repo: repoTarget };
+}
+
+/**
+ * Returns all the pull requests that are opened and not a draft from the list of repos / organizations.
+ *
+ * https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-only-issues-or-pull-requests
+ */
+export async function getOpenPullRequests(context: Context, targets: ReposWatchSettings) {
+  const { octokit, logger } = context;
+  const filter = [
+    ...targets.monitor.reduce<string[]>((acc, curr) => {
+      const parsedTarget = parseTarget(context, curr);
+      if (parsedTarget) {
+        return [...acc, parsedTarget.repo ? `repo:${parsedTarget.org}/${parsedTarget.repo}` : `org:${parsedTarget.org}`];
+      }
+      return acc;
+    }, []),
+    ...targets.ignore.reduce<string[]>((acc, curr) => {
+      const parsedTarget = parseTarget(context, curr);
+      if (parsedTarget) {
+        return [...acc, parsedTarget.repo ? `-repo:${parsedTarget.org}/${parsedTarget.repo}` : `-org:${parsedTarget.org}`];
+      }
+      return acc;
+    }, []),
+  ];
+  try {
+    const { data } = await octokit.rest.search.issuesAndPullRequests({
+      q: `is:pr is:open draft:false ${filter.join(" ")}`,
+    });
+    return data.items.flat();
+  } catch (e) {
+    logger.error(`Error getting open pull-requests for targets: [${filter.join(", ")}]. ${e}`);
+    return [];
```

Comment:
wouldn't it be better to throw the error?

Reply:
In the current flow, it would log that there is an error and then `Nothing to do` and follow the same flow as if there were no pull requests to check. I could instead making it bubble up the error, and send it to the kernel. My logic was that I didn't think it was too relevant because if that fails, that probably means GitHub API itself failed, not the plugin's code. On the other hand we might want to take action on failure so I'll change it.

Reply:
@whilefoo added throw and related test.